Allow to call $column1->setColumnDefinition($column2->getColumnDefinition())

VincentLanglet opened the pull request:
<!-- Fill in the relevant information below to help triage your pull request. -->

|      Q       |   A
|------------- | -----------
| Type         | pedantic
| Fixed issues | <!-- use #NUM format to reference an issue -->

#### Summary

<!-- Provide a summary of your change. -->
I'm trying to bump the phpstan level of my library and found this which can be improved.

VincentLanglet commented on src/Schema/Column.php:
```diff
@@ -190,7 +190,7 @@ public function setNotnull($notnull)
 
     /**
-     * @param mixed $default
+     * @param string|null $default
```
The default property only accept `string|null`
```php
    /** @var string|null */
    protected $_default;
```

morozov replied:
Technically you're right and I agree with this restriction but it will result in a BC break in 4.0 (which is also fine). For instance: https://github.com/doctrine/dbal/blob/bd8547f6a4a20c1a5780dd031bbf8cddbc218b59/tests/Functional/Schema/SQLServerSchemaManagerTest.php#L83

I'd like all these issues addressed in `3.4.x` before the merge up.

This would also require changes in documentation. For instance: https://github.com/doctrine/dbal/blob/f57646665700d833054d1ac597cd28a19e9f35a3/docs/en/reference/schema-representation.rst#L91-L92

And the nastiest part is that it's no longer clear what exactly the meaning of the column default is: previously its type was mixed, so it implied that it's the default _value_ (e.g. `(int) 666`). Now we are saying it can only be a string, so we imply that it's an SQL _expression_ but we don't support arbitrary SQL expressions.

VincentLanglet replied:
Should I split the PR in two then to not delay the `setColumnDefinition` phpdoc update. (which is the only one impacting me currently) ?

morozov replied:
That should work.

VincentLanglet replied:
Ok, I removed this changed and renamed the commit/PR.

Does an issue track the change needed for the default property ?

morozov replied:
I believe it’s already tracked somewhere. Unless you plan on working on it, no need to file a new one.

VincentLanglet commented on src/Schema/Column.php:
```diff
@@ -227,7 +227,7 @@ public function setPlatformOption($name, $value)
 
     /**
-     * @param string $value
+     * @param string|null $value
```
This allow `$column1->setColumnDefinition($column2->getColumnDefinition())`
And this was the only setter which it wasn't possible.

morozov replied:
This looks good.

VincentLanglet committed:
Allow to pass null to setColumnDefinition

This allow to call the setter with the
value return by the getter of another class

morozov commented:
Thanks, @VincentLanglet!
